Protocol.md

```diff
@@ -159,10 +146,7 @@ and closing transactions.
    * [`32*byte`:`temporary_contract_id`]
    * [`u64`:`total_collateral_satoshis`]
    * [`point`:`funding_pubkey`]
-   * [`spk`:`payout_spk`]
-   * [`u16`:`num_funding_inputs`]
-   * [`num_funding_inputs*funding_input`:`funding_inputs`]
-   * [`spk`:`change_spk`]
+   * [`psbt`:`partial_funding_psbt`]
```

Comment:
Should denote if this `partial_funding_psbt` should have the offerer's inputs and outputs or not.

(I think it should).

Reply:
Maybe a way to solve #18 is here and we can just have the accepter due the shuffling

Reply:
If we are having the offerer's inputs and outputs in this psbt it should be noted the they must verify they are all included correctly.

Reply:
> Should denote if this partial_funding_psbt should have the offerer's inputs and outputs or not.

I think it should not, it should be done exactly the same way as the offerer. The goal is that such partial funding PSBT can be created out of any wallet with PSBT support just by creating a transaction sending the collateral to yourself. If we need to combine with other inputs, the process will be more complicated. (that said, a DLC tool could easily translate a PSBT from a wallet in the first form, to the later form, it is just a bit more complicated)

There is also another reason: We don't want the offerer to have to validate the PSBT of the acceptor beyond what is strictly necessary. It is way better that both ther acceptor and offerer build the funding transaction by themselves, rather than the offerer not building the funding transaction, but have to verify that the acceptor did it right.

And if the offerer knows how to do it right, it could just do it himself in the first place.

The only reason it would make sense for the acceptor to add the offerer inputs, is if we wanted to let discretion on how to build the funding, up to the acceptor. So, instead of both agreeing on how to build the funding transaction, the acceptor decides how to build it, and the offerer enforces some rules upon it.
I think this is fine as well. No strong opinion on one way or the other.

Reply:
Thank you for the thorough explanation. I agree with you now, we should have the accepter only include their own inputs and outputs